helm-chart/binderhub/templates/deployment.yaml

```diff
@@ -98,7 +98,7 @@ spec:
               key: "binder.hub-token"
         {{- if .Values.config.BinderHub.auth_enabled }}
         - name: JUPYTERHUB_API_URL
-          value: {{ (print (.Values.config.BinderHub.hub_url | trimSuffix "/") "/hub/api/") }}
+          value: {{ (print (.Values.config.BinderHub.hub_url_local | default .Values.config.BinderHub.hub_url | trimSuffix "/") "/hub/api/") }}
```

Comment:
Nice catch adding this as well!

Reply:
If auth_enabled, this URL is required to be mounted as an env var? Hmmm? Is this because the JupyterHub imported HubOAuth classes looks for this URL or similar?

Ah yes it seems so, this could been configured with `c.HubAuth.api_url = c.BinderHub.hub_url_local` someplace instead, because JUPYTERHUB_API_URL serves the purpose of a default value for the c.HubAuth.api_url traitlet.

It seems like...

- JUPYTERHUB_API_URL can be configured with c.HubAuth.api_url
- JUPYTERHUB_BASE_URL can be configured with c.HubAuth.hub_prefix
- JUPYTERHUB_CLIENT_ID can be configured with c.HubOAuth.oauth_client_id
- JUPYTERHUB_OAUTH_CALLBACK_URL can be configured with c.HubOAuth.oauth_redirect_uri

Well, okay, let's configure them through environment variables.

Reply:
@consideRatio I was confused by the apparent duplication between config and env-vars too. I only discovered the problem when deploying it to a test system.